wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
+                        # y -= sample_medians
+                        
+                        ################################
+                        # Subtract off median between `ma` and `mb` if there is no SPE between `ma` and `mb`, else, subtract off the total median in this range using `no_SPE_trials_mask`.
+                        ################################
+                        # ma = -25
+                        # mb = 200
+                        # m_mask = np.logical_and(x>=ma, x<mb)
+                        # no_SPE_trials_mask = np.min(y[:, m_mask], axis=-1) > -2 * iqr(high_filter_y[:, m_mask].flatten())
+                        # y -= np.array([np.median(y[i, m_mask]) if no_SPE_trials_mask[i] else np.median((y[no_SPE_trials_mask, :])[:, m_mask]) for i in range(len(y))])[:, np.newaxis]
+                      
+                        ################################
+                        # For each event, subtract off the median of all points that lie above `cutoff`. Then, across all events, subtract off the median of all points between `a` and `b`.
+                        ################################
+                        cutoff = -2*iqr(high_filter_y.flatten())
+                        no_SPE_mask = y > cutoff 
+                        y -= np.array([np.median(y[i, no_SPE_mask[i]]) for i in range(len(y))])[:, np.newaxis]
+                        i_mask = np.logical_and(x>=args.s, x<args.s+args.IT)
+                        y -= np.median(y[:, i_mask])
+
+                        ################################
+                        # Delete all trials that have an SPE between `ma` and `mb`
+                        ################################
+                        # ma = -25
+                        # mb = 200
+                        # SPE_trials = np.min(y[:, np.logical_and(x >= ma, x < mb)], axis=-1) < -2 * iqr(high_filter_y[:, np.logical_and(x >= ma, x < mb)].flatten())
+                        # SPE_trials_idx = [i for i in range(len(y)) if SPE_trials[i]]
+                        # y = np.delete(y, SPE_trials_idx, axis=0)
+                        # # Subtract off the median between `ma` and `mb` per event
+                        # y -= np.median(y[:, np.logical_and(x >= ma, x < mb)], axis=-1)[:, np.newaxis]  # per event median subtraction
+                        # if len(y) == 0:
+                        #     print('All trials were removed')
+
+                        if channel in charge:
+                            charge[channel] = np.concatenate((charge[channel],integrate(x,y, a, b)))
+                            f_charge[channel] = np.concatenate((f_charge[channel], integrate(x, high_filter_y, a, b))) 
+                            # average[channel] = np.concatenate((average[channel], np.median(y, axis=-1))) 
+                            # std[channel] = np.concatenate((std[channel], np.std(y, axis=-1)))
```

Comment:
See above. Either we should just always store these or have a separate branch used for testing.